xiterator constructor missing?

When trying `auto itpair = std::minmax_element(arr.begin(), arr.end());` or similar such functions, it doesn't compile as an temporary xiterator cannot be instantiated from an empty initializer list, which `minmax_element` apparently tries.

E.g.

```
                 from /home/wolfv/Programs/xorig/examples/../include/xtensor/xutils.hpp:18,
                 from /home/wolfv/Programs/xorig/examples/../include/xtensor/xexpression.hpp:16,
                 from /home/wolfv/Programs/xorig/examples/../include/xtensor/xscalar.hpp:16,
                 from /home/wolfv/Programs/xorig/examples/tutorial.cxx:2:
/usr/include/c++/6.3.1/bits/stl_algo.h: In instantiation of ‘constexpr std::pair<_FIter, _FIter> std::__minmax_element(_ForwardIterator, _ForwardIterator, _Compare) [with _ForwardIterator = xt::xiterator<xt::xstepper<xt::xarray<double> >, std::vector<long unsigned int> >; _Compare = __gnu_cxx::__ops::_Iter_less_iter]’:
/usr/include/c++/6.3.1/bits/stl_algo.h:3409:35:   required from ‘constexpr std::pair<_FIter, _FIter> std::minmax_element(_FIter, _FIter) [with _FIter = xt::xiterator<xt::xstepper<xt::xarray<double> >, std::vector<long unsigned int> >]’
/home/wolfv/Programs/xorig/examples/tutorial.cxx:17:54:   required from here
/usr/include/c++/6.3.1/bits/stl_algo.h:3337:30: error: no matching function for call to ‘xt::xiterator<xt::xstepper<xt::xarray<double> >, std::vector<long unsigned int> >::xiterator(<brace-enclosed initializer list>)’
       _ForwardIterator __min{}, __max{};
```

--- a/include/xtensor/xgenerator.hpp
+++ b/include/xtensor/xgenerator.hpp
@@ -140,6 +140,7 @@ namespace xt
         using shape_type = typename xgenerator_type::shape_type;
         using index_type = get_index_type<shape_type>;
 
+        xgenerator_stepper() = default;
         xgenerator_stepper(const xgenerator_type* func, const shape_type& shape) noexcept;
 
         void step(size_type dim, size_type n = 1);

--- a/include/xtensor/xindexview.hpp
+++ b/include/xtensor/xindexview.hpp
@@ -198,6 +198,7 @@ namespace xt
         using shape_type = S;
         using index_type = get_index_type<shape_type>;
 
+        xindexview_stepper() = default;
         xindexview_stepper(view_type* func, const shape_type& shape) noexcept;
 
         reference operator*() const;

--- a/include/xtensor/xiterator.hpp
+++ b/include/xtensor/xiterator.hpp
@@ -83,6 +83,7 @@ namespace xt
         using size_type = typename container_type::size_type;
         using shape_type = typename container_type::shape_type;
 
+        xstepper() = default;
         xstepper(container_type* c, subiterator_type it, size_type offset) noexcept;
 
         reference operator*() const;
@@ -138,6 +139,7 @@ namespace xt
         using shape_type = S;
         using index_type = get_index_type<shape_type>;
 
+        xiterator() = default;
         xiterator(It it, const shape_type& shape);
 
         self_type& operator++();

--- a/include/xtensor/xview.hpp
+++ b/include/xtensor/xview.hpp
@@ -241,6 +241,7 @@ namespace xt
 
         using shape_type = typename substepper_type::shape_type;
 
+        xview_stepper() = default;
         xview_stepper(view_type* view, substepper_type it,
                       size_type offset, bool end = false);
 

